README.md

```diff
@@ -12,69 +12,42 @@ issuance and validation.
 and with [Microsoft Active Directory Federation Services](http://en.wikipedia.org/wiki/Active_Directory_Federation_Services).
 
-## Security Vulnerability in Versions < 1.4.6 and 2.0.0
+## 1. Security Vulnerability in Versions < 1.4.6 and 2.0.0
 _passport-azure-ad_ has a known security vulnerability affecting versions <1.4.6 and 2.0.0. Please update to >=1.4.6 or >=2.0.1 immediately. For more details, see the [security notice](https://github.com/AzureAD/passport-azure-ad/blob/master/SECURITY-NOTICE.MD).
 
-## Versions
-Current version - 3.0.0  
+## 2. Versions
+Current version - 3.0.1  
 Minimum  recommended version - 1.4.6  
 You can find the changes for each version in the [change log](https://github.com/AzureAD/passport-azure-ad/blob/master/CHANGELOG.md).
 
-## Contribution History
+## 3. Contribution History
 
 [![Stories in Ready](https://badge.waffle.io/AzureAD/passport-azure-ad.png?label=ready&title=Ready)](https://waffle.io/AzureAD/passport-azure-ad)
 
 [![Throughput Graph](https://graphs.waffle.io/AzureAD/passport-azure-ad/throughput.svg)](https://waffle.io/AzureAD/passport-azure-ad/metrics)
 
-## Installation
+## 4. Installation
 
     $ npm install passport-azure-ad
 
-## Usage
+## 5. Usage
 
-### Configure strategy
+This library contains two strategies: OIDCStrategy and BearerStrategy.
 
-This sample uses the OAuth2Bearer Strategy:
+OIDCStrategy uses OpenID Connect protocol for web application login purposes. It works in the following manner:
+If a user is not logged in, passport sends an authentication request to AAD (Azure Active Directory), and AAD prompts the user for his or her sign-in credentials. On successful authentication, depending on the flow you choose, web application will eventually get an id_token back either directly from the AAD authorization endpoint or by redeeming a code at the AAD token endpoint. Passport then validates the id_token and propagates the claims in id_token back to the verify callback, and let the framework finish the remaining authentication procedure. If the whole process is successful, passport adds the user information to `req.user` and passes it to the next middleware. In case of error, passport either sends back an unauthorized response or redirects the user to the page you specified (such as homepage or login page).
 
-```javascript
+BearerStrategy uses Bearer Token protocol to protect web resource/api. It works in the following manner:
+User sends a request to the protected web api which contains an access_token in either the authorization header or body. Passport extracts and validates the access_token, and propagates the claims in access_token to the verify callback and let the framework finish the remaining authentication procedure. On successful authentication, passport adds the user information to `req.user` and passes it to the next middleware, which is usually the business logic of the web resource/api. In case of error, passport sends back an unauthorized response.
 
```

Comment:
Should we mention  sessions here explicitly ? 
In both cases session support is optional, 
but provided description implies that session in supported for OIDC  and not supported for bearer strategy.

For OIDCStrategy  we might add that 
common scenario is accessing a web application via a browser, 
So, after successful authentication, Passport will establish a persistent login session.


Typical use case for Berarer startegy is protection of some web API where   credentials to be supplied with each request, so parameter (session : false) should be used for passport.authenticate().
What if it is set to true ? I guess token  will not be validated if there is a session. Is  (session: true) real use case for bearer strategy ?